src/Order.Data/Specifications/OrderSummarySpec/OrderSummaryByIdSpecification.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Linq;
+
+namespace Order.Data.Specifications.OrderSummarySpec
+{
+    public class OrderSummaryByIdSpecification : OrderSummarySpecification
+    {
+        public OrderSummaryByIdSpecification(Guid orderId, bool isInMemoryDatabase) : base()
+        {
+            var orderIdBytes = orderId.ToByteArray();
+            Query = x => isInMemoryDatabase ? x.Id.SequenceEqual(orderIdBytes) : x.Id == orderIdBytes;
```

Comment:
I don't like this difference for queries in tests and in production code. I think it might be fixed by using conversion in context class (see my comment in that class). But I was not able to confirm that or implement it in time.